tests/tracing/processor/test_mlflow_v3_processor.py

```diff
@@ -87,6 +87,66 @@ def test_on_start_during_run(monkeypatch):
         assert trace.info.request_metadata[TraceMetadataKey.SOURCE_RUN] == run.info.run_id
 
 
+def test_incremental_span_name_deduplication():
+    """Test that span names are deduplicated incrementally as spans end."""
+    InMemoryTraceManager.reset()
+    trace_manager = InMemoryTraceManager.get_instance()
+
+    trace_id = 12345
+    request_id = "tr-" + encode_trace_id(trace_id)
+    processor = MlflowV3SpanProcessor(span_exporter=mock.MagicMock())
+
+    # Helper to create and register a span
+    def create_and_register(name, span_id, parent_id=1):
+        span = create_mock_otel_span(
+            name=name,
+            trace_id=trace_id,
+            span_id=span_id,
+            parent_id=parent_id,
+            start_time=span_id * 1_000_000,
+            end_time=(span_id + 1) * 1_000_000,
+        )
+        processor.on_start(span)
+        live_span = LiveSpan(span, request_id)
+        live_span._request_id = request_id
+        trace_manager.register_span(live_span)
+        return span
+
+    # Create root and 4 child spans: 3 "process" and 2 "query"
+    root = create_and_register("process", 1, parent_id=None)
+    child1 = create_and_register("process", 2)
+    child2 = create_and_register("query", 3)
+    child3 = create_and_register("process", 4)
+    child4 = create_and_register("query", 5)
+
+    # End child1 - should deduplicate first two "process" spans
+    processor.on_end(child1)
+    with trace_manager.get_trace(request_id) as trace:
+        names = [s.name for s in trace.span_dict.values()]
+        assert "process_1" in names
+        assert "process_2" in names
+
+    # End child3 - should correctly number third "process" as process_3
+    processor.on_end(child3)
+    with trace_manager.get_trace(request_id) as trace:
+        names = [s.name for s in trace.span_dict.values()]
+        assert "process_3" in names  # Correctly numbered due to _original_name
+
+    # End child4 - should deduplicate "query" spans
+    processor.on_end(child4)
+    with trace_manager.get_trace(request_id) as trace:
+        names = sorted([s.name for s in trace.span_dict.values()])
```

Comment:
do we need sorted?

Reply:
We need to sort the dict spans on timestamp to make sure that the naming order is correct (which was not happening here before) - fixed!